Casks/calhash.rb

```diff
@@ -0,0 +1,34 @@
+cask "calhash" do
+  if MacOS.version <= :catalina
+    version "1.0.5"
+    sha256 "bb15c18839837bdeda203a8a916b09b4f2b971e5bba10b603f290130939d3bf7"
+  elsif MacOS.version <= :big_sur
+    version "1.1.1"
+    sha256 "66b2de6f514119b3ddf915310b94486d24d3cc6733c74af97e367b76cb2d7478"
+  else
+    version "1.1.7"
+    sha256 "bbcbbc3d3f47751d5f021603691404e0886ae999e85812c2fe55b0e98d104a04"
+  end
+
+  url "https://www.titanium-software.fr/download/#{MacOS.version.to_s.delete(".")}/CalHash.dmg"
+  name "CalHash"
+  desc "Utility that you can use to calculate and compare the checksum for a file"
+  homepage "https://www.titanium-software.fr/en/calhash.html"
+
+  livecheck do
+    url :homepage
+    regex(/>\s*CalHash\s+v?(\d+(?:\.\d+)+)\s+for\s+[\w\s.-]*\s+#{MacOS.version}\s*</i)
```

Comment:
```suggestion
    regex(/>\s*CalHash\s+v?(\d+(?:\.\d+)+)\s+for/i)
```
The vendor says that legacy versions won't be updated. So just checking with something like this should suffice.

Reply:
You are welcome! To fix the now failing livechecks for the legacy versions, use `skip` e.g. seen in `monitorcontrol`. I noticed that the other casks of that vendor do it the way you had it first, but while this satisfies CI, it isn't correct I believe. If somebody on e.g. Big Sur would run livecheck, `#{MacOS.version}` for them would be `11`, which would never signal the availability of a new version. While correct for their OS version, it isn't correct for `livecheck` purposes, as they could very well bump the cask.